core/commands/diff.py

```diff
@@ -388,52 +398,57 @@ def run(self, edit, reset=False):
             set((self.view.size() + 1, ))
         ))
 
-        self.apply_diffs_for_pts(cursor_pts, reset)
+        filter_ = partial(filter, None)
+        flatten = chain.from_iterable
 
-    def apply_diffs_for_pts(self, cursor_pts, reset):
-        in_cached_mode = self.view.settings().get("git_savvy.diff_view.in_cached_mode")
+        patches = unique(flatten(filter_(self.head_and_hunk_for_pt(pt) for pt in cursor_pts)))
```

Comment:
Correct me if I'm wrong but I believe we can simply use a set (or set comprehension), as `head_and_hunk_for_pt` returns either None or substr (which is [guaranteed to be a string][1]). What do you say?

[1]: https://www.sublimetext.com/docs/3/api_reference.html#sublime.View

Reply:
Hm, which parts of this line sould be replaceable? btw: `head_and_hunk_for_pt -> Optional[Tuple[str, str]]`

So we go:

```
			List[Optional[Tuple[str, str]]] 
- filter -> List[Tuple[str, str]]
- flat   -> List[str]
```

Reply:
Yea tuples of strings are usable in set just fine. I'm suggesting:
```
patches = ''.join(set(flatten(filter_(self.head_and_hunk_for_pt(pt) for pt in cursor_pts))))
```

But now that I think of it, it might not be good to jumble up the order of the strings; at the very least the head and hunk should come together ? I think we can scrape my suggestion in that case.

Reply:
Yes for the outer unique, we must keep everything ordered.

Reply:
Basically

```
HEAD_A, HUNK_B, HEAD_A, HUNK_C, ...
->
HEAD_A, HUNK_B, HUNK_C
```